subsys/bluetooth/controller/ll_sw/ull.c

```diff
@@ -1722,13 +1747,70 @@ static inline int rx_demux_rx(memq_link_t *link, struct node_rx_hdr *rx)
 	return 0;
 }
 
+#if defined(CONFIG_BT_CTLR_ADV_EXT) && defined(CONFIG_BT_BROADCASTER)
+static void term_evt_op_stop_cb(uint32_t status, void *param)
+{
+	static memq_link_t link;
+	static struct mayfly mfy = {0, 0, &link, NULL, NULL};
+	struct ll_adv_set *adv;
+	struct ull_hdr *hdr;
+	uint32_t ret;
+
+	/* Ignore if race between thread and ULL */
+	if (status != TICKER_STATUS_SUCCESS) {
+		/* TODO: detect race */
+
+		return;
+	}
+	adv = param;
+	hdr = &adv->ull;
+	mfy.param = &adv->lll;
+
+	if (hdr->ref) {
+		LL_ASSERT(!hdr->disabled_cb);
+		hdr->disabled_param = NULL;
+		hdr->disabled_cb = NULL;
+
+		mfy.fp = lll_disable;
+		ret = mayfly_enqueue(TICKER_USER_ID_ULL_LOW,
+				     TICKER_USER_ID_LLL, 0, &mfy);
+		LL_ASSERT(!ret);
+	}
+}
+
+static void term_evt_disable(struct ull_hdr *ull_hdr)
+{
+	struct lll_adv *lll;
+	struct ll_adv_set *adv;
+	uint8_t handle;
+	uint32_t ret;
+
+	lll = (struct lll_adv *)HDR_ULL2LLL(ull_hdr);
+	adv = (void *)HDR_LLL2EVT(lll);
+
+	handle = ull_adv_handle_get(adv);
+	LL_ASSERT(handle < BT_CTLR_ADV_SET);
```

Comment:
This check is not needed as long as ull_hdr supplied by caller is valid, instead add checks in caller if needed (helps in debugging caller).